martin/src/pg/pool.rs

```diff
@@ -42,25 +48,39 @@ impl PgPool {
 
         let conn = get_conn(&pool, &id).await?;
         let pg_ver = get_postgres_version(&conn).await?;
-        if pg_ver < MINIMUM_POSTGRES_VER {
-            return Err(PostgresqlTooOld(pg_ver, MINIMUM_POSTGRES_VER));
-        }
-        if pg_ver < RECOMMENDED_POSTGRES_VER {
-            warn!("PostgreSQL {pg_ver} is older than the recommended {RECOMMENDED_POSTGRES_VER}.");
+        if pg_ver < MINIMUM_POSTGRES_VERSION {
+            return Err(PostgresqlTooOld(pg_ver, MINIMUM_POSTGRES_VERSION));
         }
 
         let postgis_ver = get_postgis_version(&conn).await?;
-        if postgis_ver < MINIMUM_POSTGIS_VER {
-            return Err(PostgisTooOld(postgis_ver, MINIMUM_POSTGIS_VER));
+        if postgis_ver < MINIMUM_POSTGIS_VERSION {
+            return Err(PostgisTooOld(postgis_ver, MINIMUM_POSTGIS_VERSION));
+        }
+
+        // In the warning cases below, we could technically run.
+        // This is not ideal for reasons explained in the warnings
+
+        if pg_ver < RECOMMENDED_POSTGRES_VERSION {
+            warn!(
+                "PostgreSQL {pg_ver} is older than the recommended minimum {RECOMMENDED_POSTGRES_VERSION}."
+            );
+        }
+        let margin_not_supported = postgis_ver < ST_TILE_ENVELOPE_POSTGIS_VERSION;
```

Comment:
for clarity, lets also use a positive value here - otherwise below you end up with a confusing double negative `!margin_not_supported`. Also means you can use var as is when creating an instance.

```suggestion
        let supports_tile_margin = postgis_ver >= ST_TILE_ENVELOPE_POSTGIS_VERSION;
```